p2p/src/network/scheduler/p2p_network_scheduler_reducer.rs

```diff
@@ -64,148 +132,480 @@ impl P2pNetworkSchedulerState {
+                                        }
+                                    }
                                 }
                             }
-                            token::StreamKind::Broadcast(_) => {}
-                            token::StreamKind::Identify(_) => {}
-                            token::StreamKind::Discovery(_) => {}
-                            token::StreamKind::Ping(_) => {}
-                            token::StreamKind::Bitswap(_) => {}
-                            token::StreamKind::Status(_) => {}
                         }
                     }
-                    None => {}
                 }
+                Ok(())
             }
-            P2pNetworkSchedulerAction::SelectError { .. } => {
-                // NOOP, error should be triggered
+            P2pNetworkSchedulerAction::SelectError { addr, kind, error } => {
+                let dispatcher = state_context.into_dispatcher();
+
+                match kind {
+                    SelectKind::Stream(peer_id, stream_id)
+                        if keep_connection_with_unknown_stream() =>
+                    {
+                        warn!(meta.time(); summary="select error for stream", addr = display(addr), peer_id = display(peer_id));
+                        // just close the stream
+                        dispatcher.push(P2pNetworkYamuxAction::OutgoingData {
+                            addr: *addr,
+                            stream_id: *stream_id,
+                            data: Data::default(),
+                            flags: YamuxFlags::RST,
+                        });
+                        dispatcher.push(P2pNetworkSchedulerAction::PruneStream {
+                            peer_id: *peer_id,
+                            stream_id: *stream_id,
+                        });
+                    }
+                    _ => {
+                        dispatcher.push(P2pNetworkSchedulerAction::Error {
+                            addr: *addr,
+                            error: P2pNetworkConnectionError::SelectError,
+                        });
+                    }
+                }
+
+                dispatcher.push(P2pNetworkSchedulerEffectfulAction::SelectError {
+                    addr: *addr,
+                    kind: *kind,
+                    error: error.to_owned(),
+                });
+
+                Ok(())
             }
             P2pNetworkSchedulerAction::YamuxDidInit {
                 addr,
                 message_size_limit,
-                ..
+                peer_id,
             } => {
-                if let Some(cn) = self.connections.get_mut(addr) {
-                    if let Some(P2pNetworkConnectionMuxState::Yamux(yamux)) = &mut cn.mux {
-                        yamux.init = true;
-                        yamux.message_size_limit = *message_size_limit;
-                    }
+                let Some(cn) = scheduler_state.connections.get_mut(addr) else {
+                    return Ok(());
```

Comment:
Can this happen? Filter it in enabling condition?

Reply:
and bug_condition in the else it is not supposed to happen